spec/grape/validations/validators/default_spec.rb

```diff
@@ -99,6 +99,12 @@ def app
     expect(last_response.body).to eq({ array: [{ name: 'name', with_default: 'default' }, { name: 'name2', with_default: 'bar2' }] }.to_json)
   end
 
+  it 'sets default values for grouped arrays with index' do
+    get('/array?array[0][name]=name&array[1][name]=name2&array[0][with_default]=bar1')
+    expect(last_response.status).to eq(200)
+    expect(last_response.body).to eq({ array: { 0 => { name: 'name', with_default: 'bar1' }, 1 => { name: 'name2', with_default: 'default' } } }.to_json)
+  end
```

Comment:
This is a little weak on specs. Add separate tests for simpler scenarios as well.